Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Author: Alexander Richardson <alex.richardson@gmx.de>
Description: This patch makes okteta build on big endian architectures, also
 avoids installing unnecessary files for those architectures.
Last-Update: 2012-06-16

--- a/okteta/kasten/controllers/CMakeLists.txt
+++ b/okteta/kasten/controllers/CMakeLists.txt
@@ -430,6 +430,15 @@ set( OKTETA_KASTEN_CONTROLLERS_SRCS
   ${STRUCTVIEW_SRCS}
 )
 
+if( NOT NO_STRUCTURES_TOOL )
+set( STRUCTVIEW_HDRS
+  view/structures/structtool.h
+  view/structures/structtoolview.h
+  view/structures/structurestoolfactory.h
+  view/structures/structurestoolviewfactory.h
+)
+endif( NOT NO_STRUCTURES_TOOL )
+
 set( okteta_kastencontrollers_LIB_HDRS
   oktetakastencontrollers_export.h
   view/print/printcontroller.h
@@ -477,10 +486,6 @@ set( okteta_kastencontrollers_LIB_HDRS
   view/stringsextract/stringsextracttoolviewfactory.h
   view/search/searchcontroller.h
   view/search/searchuserqueryable.h
-  view/structures/structtool.h
-  view/structures/structtoolview.h
-  view/structures/structurestoolfactory.h
-  view/structures/structurestoolviewfactory.h
   document/info/documentinfotoolview.h
   document/info/documentinfotool.h
   document/overwriteonly/overwriteonlycontroller.h
@@ -518,7 +523,13 @@ install( FILES ${okteta_kastencontroller
   DESTINATION ${OKTETAKASTEN_INCLUDE_INSTALL_DIR}
   COMPONENT Devel
 )
+if( NOT NO_STRUCTURES_TOOL )
+install( FILES ${STRUCTVIEW_HDRS}
+  DESTINATION ${OKTETAKASTEN_INCLUDE_INSTALL_DIR}
+  COMPONENT Devel
+)
 install( FILES view/structures/kns/okteta-structures.knsrc  DESTINATION  ${CONFIG_INSTALL_DIR} )
+endif( NOT NO_STRUCTURES_TOOL )
 
 set( elf_structure_FILES
   view/structures/examples/elf/elf.desktop
